arelle/ValidateXbrlDTS.py

```diff
@@ -46,6 +46,7 @@
 standard_roles_other = ("xbrl.5.1.3", ())
 
 inlineDisplayNonePattern = re.compile(r"display\s*:\s*none")
+illegalXsdPatternEscapeChar = re.compile(r"\\[^nrt\\.^?*+{}()[\]pPsSiIcCdDwW-]")
```

Comment:
This matches `\|` but shouldn't (i.e., `<pattern value='\|'> matching literal `|` should be allowed).

This matches `\\z` but shouldn't (i.e., `<pattern value='\\z'>` matching the literal `\z` should be allowed).

Reply:
See https://www.w3.org/TR/2011/CR-xmlschema11-2-20110721/datatypes.diff-wd.html#cces
As I read the spec:
  * \\z  is not allowed
  * \\| is not allowed

Reply:
@hermfischer-wf I agree `\z` is not allowed, but `\\z` should be (but this code incorrectly rejects it).